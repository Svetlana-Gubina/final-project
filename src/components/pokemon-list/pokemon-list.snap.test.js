import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import {createMemoryHistory} from 'history';
import * as reactRedux from 'react-redux';
import PokemonList from './pokemon-list';

const mockStore = configureStore({});
let history;
let store;

const pokemonsList = [
  {
    "name": `bulbasaur`,
    "id": 1
  },
  {
    "name": `ivysaur`,
    "id": 2
  },
  {
    "name": `venusaur`,
    "id": 3
  },
  {
    "name": `charmander`,
    "id": 4
  },
  {
    "name": `charmeleon`,
    "id": 5
  }];

describe(`Test PokemonList`, () => {
  beforeEach(() => {
    history = createMemoryHistory();
    store = mockStore({});
  });
  it(`Should PokemonList render correctly`, () => {
    const pokemons = pokemonsList;
    const {container} = render(
        <reactRedux.Provider store={store}>
          <Router history={history}>
            <PokemonList pokemons={pokemons} />
          </Router>
        </reactRedux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});

