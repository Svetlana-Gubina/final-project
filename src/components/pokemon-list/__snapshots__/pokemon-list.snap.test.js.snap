// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PokemonList Should PokemonList render correctly 1`] = `
<div>
  <section
    class="pokemon-list"
  >
    <h2
      class="visually-hidden"
    >
      Pokemon list
    </h2>
    <div
      class=" pokemon-card"
    >
      <a
        class="pokemon-card__link"
        href="/pokemon/1"
      >
        <img
          alt="bulbasaur"
          class="pokemon-card__img"
          height="40"
          src="img/1.png"
          width="100"
        />
      </a>
      <p
        class="pokemon-card__name"
      >
        bulbasaur
      </p>
      <div
        style="display: none; color: red;"
      >
        Sorry! Something went wrong! Please try again
      </div>
      <button
        class="pokemon-card__btn"
        type="button"
      >
        Catch
      </button>
    </div>
    <div
      class=" pokemon-card"
    >
      <a
        class="pokemon-card__link"
        href="/pokemon/2"
      >
        <img
          alt="ivysaur"
          class="pokemon-card__img"
          height="40"
          src="img/2.png"
          width="100"
        />
      </a>
      <p
        class="pokemon-card__name"
      >
        ivysaur
      </p>
      <div
        style="display: none; color: red;"
      >
        Sorry! Something went wrong! Please try again
      </div>
      <button
        class="pokemon-card__btn"
        type="button"
      >
        Catch
      </button>
    </div>
    <div
      class=" pokemon-card"
    >
      <a
        class="pokemon-card__link"
        href="/pokemon/3"
      >
        <img
          alt="venusaur"
          class="pokemon-card__img"
          height="40"
          src="img/3.png"
          width="100"
        />
      </a>
      <p
        class="pokemon-card__name"
      >
        venusaur
      </p>
      <div
        style="display: none; color: red;"
      >
        Sorry! Something went wrong! Please try again
      </div>
      <button
        class="pokemon-card__btn"
        type="button"
      >
        Catch
      </button>
    </div>
    <div
      class=" pokemon-card"
    >
      <a
        class="pokemon-card__link"
        href="/pokemon/4"
      >
        <img
          alt="charmander"
          class="pokemon-card__img"
          height="40"
          src="img/4.png"
          width="100"
        />
      </a>
      <p
        class="pokemon-card__name"
      >
        charmander
      </p>
      <div
        style="display: none; color: red;"
      >
        Sorry! Something went wrong! Please try again
      </div>
      <button
        class="pokemon-card__btn"
        type="button"
      >
        Catch
      </button>
    </div>
    <div
      class=" pokemon-card"
    >
      <a
        class="pokemon-card__link"
        href="/pokemon/5"
      >
        <img
          alt="charmeleon"
          class="pokemon-card__img"
          height="40"
          src="img/5.png"
          width="100"
        />
      </a>
      <p
        class="pokemon-card__name"
      >
        charmeleon
      </p>
      <div
        style="display: none; color: red;"
      >
        Sorry! Something went wrong! Please try again
      </div>
      <button
        class="pokemon-card__btn"
        type="button"
      >
        Catch
      </button>
    </div>
  </section>
</div>
`;
